batchflow/tests/dataformat_test.py

```diff
@@ -140,41 +139,36 @@ def test_block_dataformat(self, model_and_config):
+    """ Tests in this class show that we can (or cannot) train model with given 'data_format'.
 
     There is a following pattern in every test:
-        First of all, we get 'fake_data' and 'config' via 'model_setup' fixture.
+        First of all, we get 'data' and 'config' via 'model_setup' fixture.
         Then we optionally modify 'config'. In most cases it is done only at 'location'.
-        Finally, we assert that our modification was actually applied to model by attempting
+        Finally, we assert that our modification was actually applied to a model by attempting
         to train it on a small batch.
-
     """
-    def test_last_default(self, model, model_setup, model_pipeline):
-        """ Default value for 'data_format' is 'channels_last'. """
-        fake_dataset, config = model_setup(data_format='channels_last')
-        pipeline_config = {'model_class': model, 'model_config': config}
-        test_pipeline = model_pipeline.set_config(pipeline_config)
-        total_pipeline = test_pipeline << fake_dataset
-        batch = total_pipeline.next_batch(7, n_epochs=None)
-        assert len(batch) == 7
-
-    def test_last_common(self, model, model_setup, model_pipeline):
-        """ We can explicitly pass 'data_format', it has no effect in this case. """
-        fake_dataset, config = model_setup(data_format='channels_last')
-        config['common/data_format'] = 'channels_last'
-        pipeline_config = {'model_class': model, 'model_config': config}
-        test_pipeline = model_pipeline.set_config(pipeline_config)
-        total_pipeline = test_pipeline << fake_dataset
-        batch = total_pipeline.next_batch(7, n_epochs=None)
-        assert len(batch) == 7
-
-    @pytest.mark.xfail(run=True)
-    def test_first_common(self, model, model_setup, model_pipeline):
-        """ That is intended way to communicate 'data_format' with model. """
-        fake_dataset, config = model_setup(data_format='channels_first')
-        config['common/data_format'] = 'channels_first'
-        pipeline_config = {'model_class': model, 'model_config': config}
-        test_pipeline = model_pipeline.set_config(pipeline_config)
-        total_pipeline = test_pipeline << fake_dataset
-        batch = total_pipeline.next_batch(7, n_epochs=None)
-        assert len(batch) == 7
+    @pytest.mark.parametrize('location', ['common', 'inputs/images'])
+    @pytest.mark.parametrize('data_format',
+                             [None,
+                              pytest.param('channels_first', marks=pytest.mark.xfail),
+                              'channels_last'])
+    def test_data_format(self, model, model_setup, pipeline, location, data_format):
+        """ We can explicitly pass 'data_format' to inputs or common
+
+        Notes
+        -----
+        If `data_format` is None, use a default value.
+
+        `channels_first` does not work in TF 1.12 on CPU as a corresponding pooling operation is not implemented yet.
```

Comment:
Mine actually works fine with python 3.6.8 + TF 1.12.0. 
'Default MaxPoolingOp only supports NHWC on device type CPU' is raised when I run this test locally with python 3.7.1 and TF  1.13.0rc2, and same error is raised when check by Shippable is performed.